### PR TITLE
Mark songs as ineligible for base EXP for duplicate aliases

### PR DESCRIPTION
**Pull request by Brainicism**
Also refactor `getAvailableSongCount` to use SongSelector.

### CODE REVIEW

File: src/structures/song_selector.ts [hunk 1]
```diff
@@ -225,19 +224,23 @@ export default class SongSelector {
         this.uniqueSongsPlayed.clear();
     }
 
-    getSongs(): { songs: Set<QueriedSong>; countBeforeLimit: number } {
-        if (!this.filteredSongs) {
+    getSongs(): SelectedSongs {
+        if (!this.selectedSongs) {
             return {
                 songs: new Set(),
                 countBeforeLimit: 0,
```
Comment (taahamahdi) [inline src/structures/song_selector.ts:231]: This should also have ineligibleDueToCommonAlias

File: src/structures/song_selector.ts [hunk 2]
```diff
@@ -225,19 +224,23 @@ export default class SongSelector {
...
-        return this.filteredSongs;
+        return this.selectedSongs;
     }
 
     getCurrentSongCount(): number {
-        return this.filteredSongs ? this.filteredSongs.songs.size : 0;
+        return this.selectedSongs ? this.selectedSongs.songs.size : 0;
+    }
+
+    getSelectedSongsMetadata(): SelectedSongs | null {
```
Comment (taahamahdi) [inline src/structures/song_selector.ts:242]: This function reimplements getSongs

File: src/structures/song_selector.ts [hunk 3]
```diff
@@ -571,13 +574,50 @@ export default class SongSelector {
...
+                allNamesAndAliases.map((name) =>
+                    normalizePunctuationInName(name),
+                ),
+            );
+
+            return Array.from(normalizedDeduped);
+        });
+
+        const aliasToCountMapping: { [alias: string]: number } = {};
+        for (let i = 0; i < songAliasByMatchedSong.length; i++) {
```
Comment (taahamahdi) [inline src/structures/song_selector.ts:595]: Nit: Can be replaced with for of loop

**Review by taahamahdi — CHANGES_REQUESTED** [review]

File: src/structures/song_selector.ts (hunk 1 above)
Comment (Brainicism) [reply src/structures/song_selector.ts:231]: its nullable

**Review by taahamahdi — APPROVED** [review]

**Pull request merged**